app/assets/javascripts/admin/products/services/variant_unit_manager.js.coffee

```diff
@@ -62,6 +57,14 @@ angular.module("admin.products").factory "VariantUnitManager", (availableUnits)
       (parseFloat(scale) for scale in Object.keys(@units[unitType])).sort (a, b) ->
          a - b
 
+    @availableUnitScales: (unitType) ->
+      available = availableUnits.split(",")
+      units = @units
```

Comment:
It's because inside the scope of the function passed to `filter`, `@units` isn't available.... I think your suggestion of using a parameter will take care of it though.

Reply:
in variantUnitOptions:
```ruby
available = availableUnits.split(",")
for scale in @unitScales(unit_type, available)
```

and then rewrite unitScales with something  like this;:
```ruby
@unitScales: (unitType, availableUnits = nil) ->
    scales = Object.keys(@units[unitType])
    if availableUnits
         scales = scales.filter (scale) ->
           availableUnits.includes(units[unitType][scale]['name'])

   (parseFloat(scale) for scale in scales).sort (a, b) ->
       a - b
```

Reply:
I still had to do `units = @units` but otherwise this works great: a5c5aec33a98f3b1b453cc720ef6975ce702413a

Reply:
great!
can you pls add a comment for that trick, I still dont understand why you need it :-)
I am still thinking, why is this not just: 
availableUnits.includes(@getUnitName: (scale, unitType))

Reply:
The trick was because inside the scope of the anonymous functions passed to `filter`, `@units` wasn't defined. I'm rereading it now and we could have just called `VariantUnitManager.units`, but instead, because of what you pointed out, we can use `VariantUnitManager.getUnitName` instead, and no more need for the trick. Thanks 🎉